AtalaPrismSDK/Pollux/Sources/Models/AnonCreds/AnoncredsCredentialStack+ProvableCredential.swift

```diff
@@ -0,0 +1,83 @@
+import Domain
+import Foundation
+
+extension AnoncredsCredentialStack: ProvableCredential {
+    func presentation(
+        request: Message,
+        options: [CredentialOperationsOptions]
+    ) throws -> String {
+        let requestStr: String
+        guard let attachment = request.attachments.first else {
+            throw PolluxError.messageDoesntProvideEnoughInformation
+        }
+        switch attachment.data {
+        case let attachmentData as AttachmentJsonData:
+            requestStr = try attachmentData.data.toString()
+        case let attachmentData as AttachmentBase64:
+            guard let data = Data(fromBase64URL: attachmentData.base64) else {
+                throw PolluxError.messageDoesntProvideEnoughInformation
+            }
+            requestStr = try data.toString()
+        default:
+            throw PolluxError.messageDoesntProvideEnoughInformation
+        }
+
+        guard
+            let linkSecretOption = options.first(where: {
+                if case .linkSecret = $0 { return true }
+                return false
+            }),
+            case let CredentialOperationsOptions.linkSecret(_, secret: linkSecret) = linkSecretOption
+        else {
+            throw PolluxError.missingAndIsRequiredForOperation(type: "LinkSecret")
+        }
+
+        if
+            let zkpParameters = options.first(where: {
+                if case .zkpPresentationParams = $0 { return true }
+                return false
+            }),
+            case let CredentialOperationsOptions.zkpPresentationParams(attributes, predicates) = zkpParameters
+        {
+            return try AnoncredsPresentation().createPresentation(
+                stack: self,
+                request: requestStr,
+                linkSecret: linkSecret,
+                attributes: attributes,
+                predicates: predicates
+            )
+        } else {
+            return try AnoncredsPresentation().createPresentation(
```

Comment:
Small comment or suggestion but is this if condition or else really needed? 
computeAttributes and computePredicates will not process much if the arrays or object with attributes or predicates are empty right?

Could be wrong :)

Reply:
Yeah so the way I envision this logic was.

If the user provides ZKP Parameters then we don't do anything, if they wish to provide ZKP parameters with no attributes or predicates, its their option, so we don't compute them.

If the user doesn't provide ZKP parameters then we compute a "default" version based on the presentation request.